elastica/joint.py

```diff
@@ -771,20 +803,66 @@ def _prune_using_aabbs_rod_rod(
 
 class ExternalContact(FreeJoint):
     """
-    Assumes that the second entity is a rigid body for now, can be
-    changed at a later time
+    This class is for applying contact forces between rod-cylinder and rod-rod.
+    If you are want to apply contact forces between rod and cylinder, first system is always rod and second system
+    is always cylinder.
+    In addition to the contact forces, user can define apply friction forces between rod and cylinder that
+    are in contact. For details on friction model refer to this `paper <https://www10.cs.fau.de/publications/papers/2011/Preclik_Multibody_Ext_Abstr.pdf>`_.
 
-    Most of the cylinder-cylinder contact SHOULD be implemented
-    as given in this `paper <http://larochelle.sdsmt.edu/publications/2005-2009/Collision%20Detection%20of%20Cylindrical%20Rigid%20Bodies%20Using%20Line%20Geometry.pdf>`_.
+    TODO: Currently friction force is between rod-cylinder, in future implement friction forces between rod-rod.
+
+    Examples
+    --------
+    How to define contact between rod and cylinder.
+
+    >>> simulator.connect(rod, cylidner).using(
+    ...    ExternalContact,
+    ...    k=1e4,
+    ...    nu=10,
+    ...    velocity_damping_coefficient=10,
```

Comment:
maybe this one should be set higher to keep slip threshold smaller? maybe we can add the following to the notes:

"The `velocity_damping_coefficient` is set to a high value (e.g. 1e4) to minimize slip and simulate stiction (static friction), while `friction_coefficient` corresponds to the Coulombic friction coefficient."